x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/engine/Engine.java

```diff
@@ -49,25 +50,42 @@ public class Engine implements Writeable, ToXContentObject {
      * @param name                    The name of the engine.
      * @param indices                 The list of indices targeted by this engine.
      * @param analyticsCollectionName The name of the associated analytics collection.
+     * @param updatedAt               Timestamp this Engine was last updated. Null uses current timestamp.
      */
-    public Engine(String name, String[] indices, @Nullable String analyticsCollectionName) {
+    public Engine(String name, String[] indices, @Nullable String analyticsCollectionName, Long updatedAt) {
```

Comment:
I went back and forth RE: whether to use `TimeValue` vs. `Long` here, and settled on `Long`. Open to feedback here though.

Reply:
Can we have a single ctr? Can you also add a comment regarding updatedAt, is it in seconds, milliseconds? Maybe we can name it like `updatedAtInMillis` to make it clear? Can we also use a plain `long` here?